From fb387343b7a307600a05d3a0b7bf227567543404 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Fri, 7 Jan 2011 09:46:45 -0200
Subject: [PATCH 03/37] add rhel6.1.0 machine type

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <1294393606-1424-4-git-send-email-kraxel@redhat.com>
Patchwork-id: 15940
O-Subject: [RHEL-6 kvm PATCH 3/4] add rhel6.1.0 machine type
Bugzilla: 638468
RH-Acked-by: Amit Shah <amit.shah@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

$subject says pretty much all.  This adds a new machine type
for rhel 6.1 and makes it the default because I'll need some
compat properties for rhel 6.0.

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
---
 hw/pc.c |   24 ++++++++++++++++++++++--
 1 files changed, 22 insertions(+), 2 deletions(-)

Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>
---
 hw/pc.c |   24 ++++++++++++++++++++++--
 1 files changed, 22 insertions(+), 2 deletions(-)

diff --git a/hw/pc.c b/hw/pc.c
index 3498a7c..85a4fc5 100644
--- a/hw/pc.c
+++ b/hw/pc.c
@@ -1512,6 +1512,27 @@ static void rhel_common_init(const char *type1_version,
                      strlen(buf) + 1, buf);
 }
 
+static void pc_init_rhel610(ram_addr_t ram_size,
+                            const char *boot_device,
+                            const char *kernel_filename,
+                            const char *kernel_cmdline,
+                            const char *initrd_filename,
+                            const char *cpu_model)
+{
+    rhel_common_init("RHEL 6.1.0 PC", 0);
+    pc_init_pci(ram_size, boot_device, kernel_filename, kernel_cmdline,
+                initrd_filename, cpu_model ? cpu_model : "cpu64-rhel6");
+}
+
+static QEMUMachine pc_machine_rhel610 = {
+    .name = "rhel6.1.0",
+    .alias = "pc",
+    .desc = "RHEL 6.1.0 PC",
+    .init = pc_init_rhel610,
+    .max_cpus = 255,
+    .is_default = 1,
+};
+
 static void pc_init_rhel600(ram_addr_t ram_size,
                             const char *boot_device,
                             const char *kernel_filename,
@@ -1526,11 +1547,9 @@ static void pc_init_rhel600(ram_addr_t ram_size,
 
 static QEMUMachine pc_machine_rhel600 = {
     .name = "rhel6.0.0",
-    .alias = "pc",
     .desc = "RHEL 6.0.0 PC",
     .init = pc_init_rhel600,
     .max_cpus = 255,
-    .is_default = 1,
 };
 
 static GlobalProperty compat_rhel5[] = {
@@ -1631,6 +1650,7 @@ static QEMUMachine pc_machine_rhel540 = {
 
 static void rhel_machine_init(void)
 {
+    qemu_register_machine(&pc_machine_rhel610);
     qemu_register_machine(&pc_machine_rhel600);
     qemu_register_machine(&pc_machine_rhel550);
     qemu_register_machine(&pc_machine_rhel544);
-- 
1.7.4.rc1.16.gd2f15e

